compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -70,7 +70,23 @@ impl<'tcx> ConstraintDescription for ConstraintCategory<'tcx> {
 ///
 /// Usually we expect this to either be empty or contain a small number of items, so we can avoid
 /// allocation most of the time.
-pub(crate) type RegionErrors<'tcx> = Vec<RegionErrorKind<'tcx>>;
+#[derive(Default)]
+pub(crate) struct RegionErrors<'tcx>(Vec<RegionErrorKind<'tcx>>);
+
+impl<'tcx> RegionErrors<'tcx> {
+    #[track_caller]
+    pub fn push(&mut self, val: impl Into<RegionErrorKind<'tcx>>) {
```

Comment:
you have access to tcx in the caller, could you pass that in pls? or are there too many call sites to fix?

Reply:
> you have access to tcx in the caller, could you pass that in pls? or are there too many call sites to fix?

it was pretty much me trying to keep the callers simple. I'll add it to the struct